model/qti/metadata/AbstractMetadataService.php

```diff
@@ -298,10 +298,29 @@ protected function getInstances($id, $interface)
         } else {
             foreach ($this->getOption($id) as $instance) {
                 if (is_a($instance, $interface, true)) {
-                    $this->instances[$id][] = new $instance();
+                    $this->instances[$id][] = $this->getInstance($instance);
                 }
             }
         }
+
         return $this->instances[$id];
     }
+
+    private function getInstance(string $instance): object
+    {
+        $isConfigurableService = is_a($instance, ConfigurableService::class, true);
+
+        if ($isConfigurableService) {
+            return $this->getServiceManager()->get($this->getConfigurableServiceKey($instance));
+        }
+
+        return new $instance();
```

Comment:
What if instance does not exist?

Reply:
That part was here previously, I just added a possibility to use here configurable instances too.
And [here](https://github.com/oat-sa/extension-tao-itemqti/pull/1705/files/ea47402610015b02ec7e863941cfd27fa87d4044#diff-6da3a8b998dc68c96d26d03960d04a262806d2462dbbb2fd8cf8d74239ed1c51R300) we check if this instance is implementing interface, so I think we can be sure, that if instance not exist - this check will return `false` for us.